### ISSUE TITLE
Read assigned to isoform but not found in transcript_models.gtf

### ISSUE THREAD

**Issue by doshirLV** [ablab/IsoQuant#76]
I am interested in looking at the expression of a specific gene (IPMK). It is found in read_assignments.tsv but not in transcript_models.gtf nor it is counted in gene_counts.tsv and transcript_counts.tsv.

`grep ENSG00000151151.6 00_pass.March2023data.read_assignments.tsv`
- **bbffd039-7d8c-4b8b-819e-a398bef4e781**    chr10   -       ENST00000373935.4       ENSG00000151151.6       unique  mono_exonic     58191517-58194777       PolyA=False; Classification=incomplete_splice_match;
- bbffd039-7d8c-4b8b-819e-a398bef4e781    chr10   -       ENST00000373935.4       ENSG00000151151.6       unique  mono_exonic     58191517-58194777       PolyA=False; Classification=incomplete_splice_match;

`grep bbffd039-7d8c-4b8b-819e-a398bef4e781 00_pass.March2023data.transcript_model_reads.tsv`
- bbffd039-7d8c-4b8b-819e-a398bef4e781    *

Maybe I am not understanding something but I want to be able to report the expression of this gene to the client. I am expecting a low expression value for this gene, only 2 reads match, so why is it not being reported with a count of 2? Also, what does the * mean in transcript_model_reads.tsv. I know that this means the reads are not being included in the transcript model construction, but why?

As a note: I used "all" for the options `--transcript_quantification --gene_quantification --splice_correction_strategy --model_construction_strategy`. This is ONT data and I am using a complete hg38 reference with a comprehensive gene annotation for the chromosomes only.

**Comment by andrewprzh** [comment]
Dear @doshirLV 

- As far as I see, it's the same read mentioned twice in `read_assignments.tsv`, so I seems you have only a single read matching to ENSG00000151151.6.
- The read is monoexonic (has no splice junctions), but the isoform consists of 6 exons. Since IsoQuant does not see any splice junctions matching, it decides that this is insufficient evidence of the isoform expression. Of course, I cannot say this logic is 100% correct, but IsoQuant prefers conservative strategy in general.
- `bbffd039-7d8c-4b8b-819e-a398bef4e781 *` means that when constructing a GTF file with expressed isoforms (novel and known), this read was not assigned to any transcripts, because ENSG00000151151.6 is not reported in the final GTF (1 read unspliced read is not enough, all splice junctions should be confirmed).

To summarize, if there is no ambiguity in read-to-isoform assignment and this read does not have ambiguous alignments, it is rather safe to say that ENSG00000151151.6 expresses with a low level. However, getting more evidence, such as confirmed splice junctions, would increase reliability of this conclusion.

Hope this helps!

Best
Andrey

**Comment by doshirLV** [comment]
Thank you for your response, Andrey,

I apologize if there was any confusion, there was another read with a different ID that matched but I had an error in copying it over correctly. Regardless, the discrepancy between the number of exons in the annotated and expressed isoform which resulted in IsoQuant deciding there is insufficient evidence makes sense. I understand why it was not counted and will be able to explain its low expression.

I appreciate your time,
Raj

**Comment by andrewprzh** [comment]
Dear @doshirLV 

I see! You are welcome!

Best 
Andrey
